plugins/plugin-webpack/plugin.js

```diff
@@ -313,6 +331,17 @@ module.exports = function plugin(config, args = {}) {
         new MiniCssExtractPlugin({
           filename: cssOutputPattern,
         }),
+        new CleanWebpackPlugin({
+          cleanOnceBeforeBuildPatterns: [
+            '**/*',
+            // Static-mounted files should be part of the output, but Webpack
+            // has no knowledge of them since they aren't processed at all, so
+            // they must be manually ignored.
+            ...mountedRoots(config.mount || {}).staticRoots.map(
+              (url) => `!${url.replace(/^\//, '')}/**`,
+            ),
```

Comment:
Technically it would probably be more accurate to ignore *all* static files, not just the files/directories at the root of the static mount, but I was concerned that that could lead to unexpectedly poor performance in deeply nested directories with many files.

Globbing `!url/**` ignores both `url` and `url/**` at once, according to [the library that clean-webpack-plugin uses for removing files](https://github.com/sindresorhus/del#beware).